tests/perf/pubsub_bulk_subscribe_http/test.js

```diff
@@ -0,0 +1,116 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import http from "k6/http";
+import { check } from "k6";
+import ws from "k6/ws";
+
+const targetUrl = __ENV.TARGET_URL
+const pubsubName = __ENV.PUBSUB_NAME
+const subscribeType = __ENV.SUBSCRIBE_TYPE
+const httpReqDurationThreshold = __ENV.HTTP_REQ_DURATION_THRESHOLD
+const defaultTopic = "perf-test"
+const defaultCount = 100
+const hundredBytesMessage = "a".repeat(100)
+
+export const options = {
+    discardResponseBodies: true,
+    thresholds: {
+        checks: ['rate==1'],
+        // 95% of requests should be below HTTP_REQ_DURATION_THRESHOLD milliseconds
+        http_req_duration: ['p(95)<' + httpReqDurationThreshold],
+    },
+    scenarios: {
+        bulkPSubscribe: {
+            executor: "ramping-vus",
+            startVUs: 0,
+            stages: [
+                { duration: "2s", target: 100 },
+                { duration: "3s", target: 300 },
+                { duration: "4s", target: 500 },
+            ],
+            gracefulRampDown: "0s",
+        },
+    },
+};
+
+const DAPR_ADDRESS = `http://127.0.0.1:${__ENV.DAPR_HTTP_PORT}`;
+
+/**
+ * Publishes messages to a topic using the bulk publish API.
+ * @param {String} pubsub pubsub component name
+ * @param {String} topic topic name
+ * @param {String} message message to publish
+ * @param {Number} count number of messages to publish
+ * @returns 
+ */
+function publishMessages(pubsub, topic, message, count) {
+    const bulkPublishBody = [];
+    for (let i = 0; i < count; i++) {
+        bulkPublishBody.push({
+            entryId: i.toString(),
+            event: message,
+            contentType: "text/plain",
+        });
+    }
+
+    return http.post(
+        `${DAPR_ADDRESS}/v1.0-alpha1/publish/bulk/${pubsub}/${topic}`,
+        JSON.stringify(bulkPublishBody),
+        { headers: { 'Content-Type': 'application/json' }, });
+}
+
+export default function () {
+    const url = `ws://${targetUrl}/test`
+    const params = { tags: { "subscribeType": subscribeType } }
+
+    const res = ws.connect(url, params, (socket) => {
+        socket.on("open", () => {
+            // Tell the application to use bulk subscribe or normal subscribe
+            socket.send(subscribeType)
+
+            // Publish messages to the topic
+            const publishResponse = publishMessages(pubsubName, defaultTopic, hundredBytesMessage, defaultCount);
+            check(publishResponse, {
+                "publish response status code is 2xx": (r) => r.status >= 200 && r.status < 300
+            });
+        });
+        socket.on("message", (data) => {
+            console.log("Received data: " + data);
+            check(data, {
+                "completed with success": (d) => d === "true",
+            })
+        });
+        socket.on("close", () => {
+            console.log("closed");
```

Comment:
you can add a counter here and check that it should be zero (if not expected)

Reply:
The app closes the socket connection after writing the test response and triggers the `close` event. So it should not be zero. Also the k6 test continues to run until the app closes the socket (since we don't have an explicit `socket.Close()` or timeout on our end).

Reply:
Added an explicit timeout and counter to check if we had to close manually.